tests/pom/signUp.spec.js

```diff
@@ -0,0 +1,127 @@
+import { expect, test } from "@playwright/test";
+import SignUpPage from "../../src/pageObjects/signUpPage.js";
+
+import { generateRandomString } from "../../src/helpers/utils.js";
+
+const commonSignUpSteps = async (signUpPage, signupData) => {
+  await signUpPage.openSignUpModal();
+  await signUpPage.signUpUser(signupData);
+};
+
+const signUpTest = async (page, signupData, expectedUrl) => {
+  const signUpPage = new SignUpPage(page);
+  await commonSignUpSteps(signUpPage, signupData);
+
+  if (expectedUrl) {
+    await signUpPage.waitForUserNavDropdown();
+    expect(page.url()).toContain(expectedUrl);
+  } else {
+    await signUpPage.waitForExistingUserError();
+    expect(await signUpPage.getExistingUserErrorText()).toContain(
+      "User already exists",
+    );
+  }
+};
```

Comment:
it makes your code over complicated. Use page object without this wrappers. If you want to perform some action before each test use `test.beforeAEach` hook